lib/deploy/deploy.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const BbPromise = require('bluebird');
+
+module.exports = {
+  /*
+   * Will get stack resources and create object and append in obj
+   * with key as ARN Name and value as ARN Value
+  */
+  getLambdaStackResource() {
+    const resources = [];
+    const stackName = this.provider.naming.getStackName(this.options.stage);
+    return this.provider.request('CloudFormation', 'describeStacks', { StackName: stackName }, this.options.stage, this.options.region).then((result) => {
+      if (result) {
+        result.Stacks[0].Outputs.forEach((output) => {
+          let outputValue = output.OutputValue;
+          outputValue = output.OutputKey.includes('Lambda') ? outputValue.substring(0, outputValue.lastIndexOf(':')) : outputValue;
+          resources.push({ [output.OutputKey]: outputValue });
+        });
+        this.deployStateMachine = {};
+        this.deployStateMachine.getLambdaStackResource = resources;
+      }
+      return BbPromise.resolve();
+    });
+  },
+  replaceAllStatesARNInDefinition(definition, value) {
+    function iterateStates(obj) {
+      const data = {};
+      data.States = {};
+      for (const key in obj) {
+        if (key === 'States') {
+          const subStates = obj[key];
```

Comment:
Please use [Object.entries](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries)

Reply:
Used and modified logic along with simplifying calling.